chore: add auth and model provider configuration documentation (#534)
otto8-ai/otto8: docs/docs/04-configuration/02-model-providers.md

# Model Providers

> Each package model provider is configured slightly differently, but, at least, the `OTTO_*_MODEL_PROVIDER_API_KEY` environment variable are required, where `*` stands in for the name of the provider (e.g. `OTTO_AZURE_OPENAI_MODEL_PROVIDER_API_KEY`).

Reviewer: ```suggestion
Each model provider has a slightly different configuration, but all require at least the `OTTO_*_MODEL_PROVIDER_API_KEY` environment variable, where `*` stands in for the name of the provider (e.g. `OTTO_AZURE_OPENAI_MODEL_PROVIDER_API_KEY`).
```

Author: Done

Below is a summary of the configuration options for each provider. However, the packaged model providers are configured to indicate which environment variables are required and which ones are not set in the current environment. For example, `/api/tool-references/azure-openai-model-provider` would indicate the status of the Azure OpenAI model provider. If the environment variables are not set, then the API would return something like:

```json
{
  "id": "azure-openai-model-provider",
  "created": "2024-11-08T16:03:21-05:00",
  "metadata": {
    "envVars": "OTTO_AZURE_OPENAI_MODEL_PROVIDER_API_KEY,OTTO_AZURE_OPENAI_MODEL_PROVIDER_ENDPOINT,OTTO_AZURE_OPENAI_MODEL_PROVIDER_DEPLOYMENT_NAME"
  },
  "name": "Azure OpenAI Provider",
  "toolType": "modelProvider",

>   "reference": "github.com/otto8-ai/tools/./azure-openai-model-provider",

Reviewer 2: `github.com/otto8-ai/tools/./azure-openai-model-provider` is this a valid reference that it must have `./` in the middle?

Author: This is the way they come out of the API. I change them in the docs for ease of understanding.
